script/import_scripts/base.rb

```diff
@@ -563,7 +563,7 @@ def create_post(opts, import_id)
     post_creator = PostCreator.new(user, opts)
     post = post_creator.create
     post_create_action.try(:call, post) if post
-    post ? post : post_creator.errors.full_messages
+    post_creator.errors.full_messages.empty? ? post : post_creator.errors.full_messages
```

Comment:
can `post_creator.errors` be `nil` ? I am pretty sure it can be. Instead:

```
if !post || post_creator.errors&.length > 0 ? post : post_creator.errors.full_messages
```

Reply:
I don't think `post_creator.errors` can be `nil`, but it certainly could be `blank`:
`PostCreator` seems to include `HasErrors` which initialises:
`@errors ||= ActiveModel::Errors.new(self)`

Within the `valid?` function, it then checks:
`valid = @post.errors.blank?`

However, your suggestion would also catch this. I've had a look to see if `full_messages` **must** be present if `errors` exist, and I think it does.
```suggestion
    if !post || post_creator.errors&.length > 0 ? post : post_creator.errors.full_messages
```